test/cli/test/bun-test.test.ts

```diff
@@ -693,6 +693,21 @@ describe("bun test", () => {
     });
     test.todo("check formatting for %p", () => {});
   });
+  // https://github.com/oven-sh/bun/issues/4118
+  test("expect NODE_ENV == 'test'", () => {
+    const stderr = runTest({
+      input: `
+        import {expect, test} from "bun:test";
+        test("expect NODE_ENV=test", () => {
+            expect(process.env.NODE_ENV).toBe("test");
+            expect(process?.env.NODE_ENV).toBe("test");
+            expect(process && process.env && process.env.NODE_ENV).toBe("test");
```

Comment:
Don't these all test the same thing? The first line is enough.

Reply:
This was the repro on the issue. They were mainly concerned with the optional chaining example.